packages/cashier/src/pages/payment-agent/payment-agent-unlisted-withdraw-form/payment-agent-unlisted-withdraw-form.jsx

```diff
@@ -0,0 +1,181 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Field, Formik, Form } from 'formik';
+import { Button, Icon, Input, Text } from '@deriv/components';
+import { getDecimalPlaces, getCurrencyDisplayCode, validNumber, website_name } from '@deriv/shared';
+import { localize, Localize } from '@deriv/translations';
+import { connect } from 'Stores/connect';
+import PaymentAgentDisclaimer from '../payment-agent-disclaimer';
+import ErrorDialog from 'Components/error-dialog';
+import SideNote from 'Components/side-note';
+import './payment-agent-unlisted-withdraw-form.scss';
+
+const validateWithdrawal = (values, { balance, currency }) => {
+    const errors = {};
+
+    const { is_ok, message } = validNumber(values.amount, {
+        type: 'float',
+        decimals: getDecimalPlaces(currency),
+    });
+
+    if (!values.amount) {
+        errors.amount = localize('This field is required.');
+    } else if (!is_ok) {
+        errors.amount = message;
+    } else if (+balance < +values.amount) {
+        errors.amount = localize('Insufficient balance.');
+    } else if (!values.account_number) {
+        errors.account_number = localize('This field is required.');
+    } else if (!/^[A-Z]{2}[A-Z\d]\d+$/.test(values.account_number)) {
+        errors.account_number = localize('Please enter a valid account number. Example: CR123456789');
+    }
+
+    return errors;
+};
+
+const PaymentAgentUnlistedWithdrawForm = ({
+    balance,
+    currency,
+    error,
+    onMount,
```

Comment:
@george-usynin-binary Just a question, I'm wondering why we need to handle this component `onMount` outside of the component itself 🤔 Why do we have some logic dependent on this component lifecycle but it’s not being handled by the component itself? 🤔
I have seen it in other places as well, Do you have any idea?

Reply:
@farzin-fs sorry, not fully understand your query. `onMount` is not a component - `onMount` is a method (action) which we receive from the payment-agent store `(onMount: modules.cashier.payment_agent.onMountPaymentAgentWithdraw)`
we use it in useEffect hook below to setup initial values for payment-agent withdraw when we landed on payment-agent withdraw screen.

Reply:
@george-usynin-binary Probably I'm misunderstanding because of the naming, I will investigate more about this and will let you know if I had more questions 🙇🏻 Thank you buddy 🌹